### PR TITLE
Process modifiers interfuses

### PR DESCRIPTION
**Pull request by alrik-durand**
## Description
I've added two interfuses for process_value and process_control interfaces.
Theses interfuses can use an interpolation function to convert the values in any way.

## Motivation and Context
Process values and control are useful for old kind of applications. But sometimes you're not interested in the value in itself but some function of it. Here is two examples that motivated me to create theses interfuses 
* Multiplying the value read from a power meter by a factor of 10 to monitor the power on the other side of a beam splitter
* Converting a simple setpoint power to a non linear voltage setpoint based on a calibration

## How Has This Been Tested?
I've checked with the demo hardware and the process_modifier_test.cfg demo config

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have documented my changes in the changelog (`documentation/changelog.md`)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added/updated for the module the config example in the docstring of the class accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

File: config/example/calib/process_modifier_test.calib [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# X Y
```
Comment (Neverhorst) [inline config/example/calib/process_modifier_test.calib]: Please do not add a new folder _calib_ just for the example. Rather define a 1:1 interpolation as default value (no change in data) and do not enforce the load from file.

File: config/example/process_modifier_test.cfg [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    processdummy:
+        module.Class: 'process_dummy.ProcessDummy'
+
+logic:
+    process_value_modifier:
+        module.Class: 'interfuse.process_value_modifier.ProcessValueModifier'
+        connect:
+            hardware: 'processdummy'
+        calibration_file: 'config/example/calib/process_modifier_test.calib'
+        force_calibration_from_file: True
```
Comment (Neverhorst) [inline config/example/process_modifier_test.cfg]: see above.

File: config/example/process_modifier_test.cfg [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    process_control_modifier:
+        module.Class: 'interfuse.process_control_modifier.ProcessControlModifier'
+        connect:
+            hardware: 'processdummy'
+        calibration_file: 'config/example/calib/process_modifier_test.calib'
+        force_calibration_from_file: True
```
Comment (Neverhorst) [inline config/example/process_modifier_test.cfg]: see above.

File: logic/interfuse/process_control_modifier.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+
+"""
+This file contains the Qudi interfuse between a process control and a process control.
```
Comment (Neverhorst) [inline logic/interfuse/process_control_modifier.py]: Guess this is some copy/paste error?

File: logic/interfuse/process_control_modifier.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                                 'given. Please update calibration before using')
+        else:
+            self._update()
+
+    def on_deactivate(self):
+        """ Deactivate module.
+        """
+        pass
+
+    def _update(self, calibration=None):
```
Comment (Neverhorst) [inline logic/interfuse/process_control_modifier.py]: Please do not make the `update` method private. You will need to call this method from external modules/scripts to change the interpolation. Also please give it a more descriptive name ("update" is a bit too generic).

File: logic/interfuse/process_value_modifier.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                                 'given. Please update calibration before using')
+        else:
+            self._update()
+
+    def on_deactivate(self):
+        """ Deactivate module.
+        """
+        pass
+
+    def _update(self, calibration=None):
```
Comment (Neverhorst) [inline logic/interfuse/process_value_modifier.py]: see above

File: logic/interfuse/process_control_modifier.py (hunk 4 above)
Comment (alrik-durand) [reply logic/interfuse/process_control_modifier.py]: Well no I don't think so. This interfuse connect to a process control and act like a process control.
Or am I missing something ?

Comment (Neverhorst) [reply logic/interfuse/process_control_modifier.py]: Ah well, yes you are right... it just looked weird.

**Pull request merged**